### PR TITLE
[AIRFLOW-6027] Disable API access by default

### PR DESCRIPTION
**Pull request by mik-laj**
Default configuration should be secure, so open unsecured API is a big problem.  A lot of Airflow instances may not change the default configuration and therefore be vulnerable to third party attacks.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6027
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: docs/security.rst [hunk 1]
```diff
@@ -113,27 +113,40 @@ alter the content and make it part of the ``PYTHONPATH`` and configure it as a b
 API Authentication
 ------------------
 
-Authentication for the API is handled separately to the Web Authentication. The default is to not
-require any authentication on the API i.e. wide open by default. This is not recommended if your
-Airflow webserver is publicly accessible, and you should probably use the ``deny all`` backend:
+Authentication for the API is handled separately to the Web Authentication. By default, access to it is
+blocked. because The authorization configuration is configured as follows.
```
Comment (feluelle) [inline docs/security.rst]:
```suggestion
blocked, because the authorization configuration is configured as follows.
```

File: docs/security.rst [hunk 2]
```diff
@@ -113,27 +113,40 @@ alter the content and make it part of the ``PYTHONPATH`` and configure it as a b
...
+
+    [api]
+    auth_backend = airflow.api.auth.backend.allow_all
+
+To enabled **password authentication**, set the following in the configuration:
+
+.. code-block:: ini
+
+    [api]
+    auth_backend = airflow.contrib.auth.backends.password_auth
```
Comment (feluelle) [inline docs/security.rst]: This block is a duplicate. See below.

File: docs/security.rst [hunk 3]
```diff
@@ -113,27 +113,40 @@ alter the content and make it part of the ``PYTHONPATH`` and configure it as a b
...
+
+To enabled **password authentication**, set the following in the configuration:
```
Comment (feluelle) [inline docs/security.rst]:
```suggestion
To enable **password authentication**, set the following in the configuration:
```

**Comment by kaxil** [comment]
Can you please add a description in the PR and JIRA on why this is needed?

**Comment by mik-laj** [comment]
I added description, but i need to find time to make Travis happy.

**Comment by ad-m** [comment]
Can we obtain CVE for that?

**Comment by mik-laj** [comment]
Do you have experience in this topic?

**Comment by ad-m** [comment]
Not, but GitHub support advisory ( https://help.github.com/en/github/managing-security-vulnerabilities/publishing-a-security-advisory ) for publishing & obtain CVE.

**Pull request closed without merging**